# Forcasting Nutrient Density of Food Consumption

## Background

## Business Understanding

### Problem Statements

1. What 

2. Why

3. How

### Goals

1. 

2. 

3. 

### Solution Statements

1. 

2.

3.

## Data Understanding

### Dataset

Resouce:
[Food Nutrition Dataset](https://www.kaggle.com/datasets/utsavdey1410/food-nutrition-dataset/data)

#### Overview

The Comprehensive Nutritional Food Database provides detailed nutritional information for a wide range of food items commonly consumed around the world. This dataset aims to support dietary planning, nutritional analysis, and educational purposes by providing extensive data on the macro and micronutrient content of foods.

#### Column Description

No | Column | Description
---|--------|------------
1 | Food | The name or type of the food item
2 | Caloric Value | Total energy provided by the food, typically measured in kilocalories(kcal) per 100 grams.
3 | Fat (in g) | Total amount of fats in grams per 100 grams, including the breakdowns that follow
4 | Saturated Fats (in g) | Amount of saturated fats (fats that typically raise the level of cholesterol in the blood) in grams per 100 grams.
5 | Monounsaturated Fats (in g) | Amount of monounsaturated fats (considered heart-healthy fats) in grams per 100 grams.
6 | Polyunsaturated Fats (in g) | Amount of polyunsaturated fats (include essential fats your body needs but can't produce itself) in grams per 100 grams.
7 | Carbohydrates (in g) | Total carbohydrates in grams per 100 grams, including sugars.
8 | Sugars (in g) | Total sugars in grams per 100 grams, a subset of carbohydrates.
9 | Protein (in g) | Total proteins in grams per 100 grams, essential for body repair and growth.
10 | Dietary Fiber (in g) | Fiber content in grams per 100 grams, important for digestive health.
11 | Cholesterol (in mg) | Cholesterol content in milligrams per 100 grams, pertinent for cardiovascular health.
12 | Sodium (in mg) | Sodium content in milligrams per 100 grams, crucial for fluid balance and nerve function.
13 | Water (in g) | Water content in grams per 100 grams, which affects the food's energy density.
14 | Vitamin A (in mg) | Amount of Vitamin A in micrograms per 100 grams, impoertant for vision and immune functioning.
15 | Vitamin B1 (Thiamine)(in mg) | Essential for glucose metabolism.
16 | Vitamin B11 (Folic Acid)(in mg) | Crucial for cell function and tissue growth, particularly important in pregnancy.
17 | Vitamin B12(in mg) | Important for brain function and blood formation.
18 | Vitamin B2 (Riboflavin)(in mg) | Necessary for energy production, cell function, and fat metabolism.
19 | Vitamin B3 (Niacin)(in mg) | Support digestive system, skin, and nerves health.
20 | Vitamin B5 (Pantothenic Acid)(in mg) | Necessary for making blood cells and helps convert food into energy.
21 | Viatmin B6 (in mg): Important for normal brain development and keeping the nervous and immune system healthy.
22 | Vitamin C (in mg) | Important for the repair of all body tissues.
23 | Vitamin D (in mg) | Crucial for absorption of calcium, promoting bone growth and health.
24 | Vitamin E (in mg) | Acts as an antioxidant, helping to protect cells from the damage caused by free radicals.
25 | Vitamin K (in mg) | Necessary for blood clotting and bone health.
26 | Calcium (in mg) | Vital for building and maintaining strong bones and teeth.
27 | Copper (in mg) | Helps with the formation of collagen, increases the absorption of iron and plays a role in energy production.
28 | Iron (in mg) | Essential for the creation of red blood cells.
29 | Magnesium (in mg) | Important for many processes in the body including regulation of muscle and nerve function, blood sugar levels, and blood pressure and making protein, bone, and DNA.
30 | Manganese (in mg) | Involved in the formation of bones, blood clotting factors, and enzymes that play a role in fat and carbohydrate metabolism, calcium absorption, and blood sugar regulation.
31 | Phosporus (in mg) | Helps with the formation of bones and teeth and is necessary for the body to make protein for the growth, maintenance, and repair of cells and tissues.
32 | Potassium (in mg) | Helps regulate fluid balance, muscle contractions, and nerve signals.
33 | Selenium (in mg) | Important for reproduction, thyroid gland function, DNA production, and protecting the body from damage caused by free radicals and from infection.
34 | Zinc (in mg) | Necessary for the immune system to properly function and plays a role in cell division, cell growth, wound healing, and the breakdown of carbohydrates.
35 | Nutrition Density | A metric indicating the nutrient richness of the food per calorie.

### Explore

#### Import Libraries

In [1]:
import numpy as np
import pandas as pd
import os 
import matplotlib.pyplot as plt
import seaborn as sns

#### Load Data

In [18]:
# Data 1
data_1_url = "https://raw.githubusercontent.com/Sulbae/OlahGizi-Project/refs/heads/main/Nutrition%20Prediction%20Model/dataset/FOOD-DATA-GROUP1.csv"
data_1_df = pd.read_csv(data_1_url)

# Data 2
data_2_url = "https://raw.githubusercontent.com/Sulbae/OlahGizi-Project/refs/heads/main/Nutrition%20Prediction%20Model/dataset/FOOD-DATA-GROUP2.csv"
data_2_df = pd.read_csv(data_2_url)

# Data 3
data_3_url = "https://raw.githubusercontent.com/Sulbae/OlahGizi-Project/refs/heads/main/Nutrition%20Prediction%20Model/dataset/FOOD-DATA-GROUP3.csv"
data_3_df = pd.read_csv(data_3_url)

# Data 4
data_4_url = "https://raw.githubusercontent.com/Sulbae/OlahGizi-Project/refs/heads/main/Nutrition%20Prediction%20Model/dataset/FOOD-DATA-GROUP4.csv"
data_4_df = pd.read_csv(data_4_url)

# Data 5
data_5_url = "https://raw.githubusercontent.com/Sulbae/OlahGizi-Project/refs/heads/main/Nutrition%20Prediction%20Model/dataset/FOOD-DATA-GROUP5.csv"
data_5_df = pd.read_csv(data_5_url)

#### Check Data

In [19]:
all_data = [data_1_df, data_2_df, data_3_df, data_4_df, data_5_df]

for data in all_data:
    print(data.shape)
    print(data.columns.tolist())

(551, 37)
['Unnamed: 0.1', 'Unnamed: 0', 'food', 'Caloric Value', 'Fat', 'Saturated Fats', 'Monounsaturated Fats', 'Polyunsaturated Fats', 'Carbohydrates', 'Sugars', 'Protein', 'Dietary Fiber', 'Cholesterol', 'Sodium', 'Water', 'Vitamin A', 'Vitamin B1', 'Vitamin B11', 'Vitamin B12', 'Vitamin B2', 'Vitamin B3', 'Vitamin B5', 'Vitamin B6', 'Vitamin C', 'Vitamin D', 'Vitamin E', 'Vitamin K', 'Calcium', 'Copper', 'Iron', 'Magnesium', 'Manganese', 'Phosphorus', 'Potassium', 'Selenium', 'Zinc', 'Nutrition Density']
(319, 37)
['Unnamed: 0.1', 'Unnamed: 0', 'food', 'Caloric Value', 'Fat', 'Saturated Fats', 'Monounsaturated Fats', 'Polyunsaturated Fats', 'Carbohydrates', 'Sugars', 'Protein', 'Dietary Fiber', 'Cholesterol', 'Sodium', 'Water', 'Vitamin A', 'Vitamin B1', 'Vitamin B11', 'Vitamin B12', 'Vitamin B2', 'Vitamin B3', 'Vitamin B5', 'Vitamin B6', 'Vitamin C', 'Vitamin D', 'Vitamin E', 'Vitamin K', 'Calcium', 'Copper', 'Iron', 'Magnesium', 'Manganese', 'Phosphorus', 'Potassium', 'Selenium

* All data have 37 typical columns
* Data Shape:
    - Data 1 has 551 rows, 
    - Data 2 has 319 rows, 
    - Data 3 has 571 rows, 
    - Data 4 has 232 rows, 
    - Data 5 has 722 rows.
* There are two unknown columns that look useless, namely columns Unnamed: 0.1 and Unnamed: 0.

In [20]:
# Drop the unkonwn columns
columns_to_drop = ["Unnamed: 0.1", "Unnamed: 0"]

for data in all_data:
    data.drop(columns=columns_to_drop, inplace=True)
    print(data.shape)
    print(data.columns.tolist())

(551, 35)
['food', 'Caloric Value', 'Fat', 'Saturated Fats', 'Monounsaturated Fats', 'Polyunsaturated Fats', 'Carbohydrates', 'Sugars', 'Protein', 'Dietary Fiber', 'Cholesterol', 'Sodium', 'Water', 'Vitamin A', 'Vitamin B1', 'Vitamin B11', 'Vitamin B12', 'Vitamin B2', 'Vitamin B3', 'Vitamin B5', 'Vitamin B6', 'Vitamin C', 'Vitamin D', 'Vitamin E', 'Vitamin K', 'Calcium', 'Copper', 'Iron', 'Magnesium', 'Manganese', 'Phosphorus', 'Potassium', 'Selenium', 'Zinc', 'Nutrition Density']
(319, 35)
['food', 'Caloric Value', 'Fat', 'Saturated Fats', 'Monounsaturated Fats', 'Polyunsaturated Fats', 'Carbohydrates', 'Sugars', 'Protein', 'Dietary Fiber', 'Cholesterol', 'Sodium', 'Water', 'Vitamin A', 'Vitamin B1', 'Vitamin B11', 'Vitamin B12', 'Vitamin B2', 'Vitamin B3', 'Vitamin B5', 'Vitamin B6', 'Vitamin C', 'Vitamin D', 'Vitamin E', 'Vitamin K', 'Calcium', 'Copper', 'Iron', 'Magnesium', 'Manganese', 'Phosphorus', 'Potassium', 'Selenium', 'Zinc', 'Nutrition Density']
(571, 35)
['food', 'Caloric 

##### Data 1

In [21]:
data_1_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 551 entries, 0 to 550
Data columns (total 35 columns):
 #   Column                Non-Null Count  Dtype  
---  ------                --------------  -----  
 0   food                  551 non-null    object 
 1   Caloric Value         551 non-null    int64  
 2   Fat                   551 non-null    float64
 3   Saturated Fats        551 non-null    float64
 4   Monounsaturated Fats  551 non-null    float64
 5   Polyunsaturated Fats  551 non-null    float64
 6   Carbohydrates         551 non-null    float64
 7   Sugars                551 non-null    float64
 8   Protein               551 non-null    float64
 9   Dietary Fiber         551 non-null    float64
 10  Cholesterol           551 non-null    float64
 11  Sodium                551 non-null    float64
 12  Water                 551 non-null    float64
 13  Vitamin A             551 non-null    float64
 14  Vitamin B1            551 non-null    float64
 15  Vitamin B11           5

In [22]:
data_1_df.head()

food  Caloric Value  ...   Zinc  Nutrition Density
0                      cream cheese             51  ...  0.039              7.070
1                 neufchatel cheese            215  ...  0.700            130.100
2  requeijao cremoso light catupiry             49  ...  0.000              5.400
3                    ricotta cheese             30  ...  0.035              5.196
4              cream cheese low fat             30  ...  0.053             27.007

[5 rows x 35 columns]

In [37]:
data_1_df.describe(include="all").T

count unique           top  ...    50%       75%     max
food                    551    551  cream cheese  ...    NaN       NaN     NaN
Caloric Value         551.0    NaN           NaN  ...  186.0     337.0  1578.0
Fat                   551.0    NaN           NaN  ...    6.6      14.4    87.5
Saturated Fats        551.0    NaN           NaN  ...    1.8       4.8    43.5
Monounsaturated Fats  551.0    NaN           NaN  ...    2.2      5.15    48.0
Polyunsaturated Fats  551.0    NaN           NaN  ...    1.1       2.6    40.1
Carbohydrates         551.0    NaN           NaN  ...    5.1      28.7   128.3
Sugars                551.0    NaN           NaN  ...    0.0       2.2    70.8
Protein               551.0    NaN           NaN  ...   11.9     25.65    86.9
Dietary Fiber         551.0    NaN           NaN  ...    0.0       1.5    17.5
Cholesterol           551.0    NaN           NaN  ...   33.9     82.65   352.5
Sodium                551.0    NaN           NaN  ...    0.4       0.9     6.1
Water                 551.0    NaN           NaN  ...   76.7    169.05   535.8
Vitamin A             551.0    NaN           NaN  ...  0.042    0.0875     2.1
Vitamin B1            551.0    NaN           NaN  ...  0.086       0.2     1.9
Vitamin B11           551.0    NaN           NaN  ...  0.051     0.083     1.3
Vitamin B12           551.0    NaN           NaN  ...  0.039     0.068     0.4
Vitamin B2            551.0    NaN           NaN  ...    0.1       0.3     3.8
Vitamin B3            551.0    NaN           NaN  ...    1.8       4.2    57.8
Vitamin B5            551.0    NaN           NaN  ...    0.5       1.2    31.4
Vitamin B6            551.0    NaN           NaN  ...    0.1       0.4     4.3
Vitamin C             551.0    NaN           NaN  ...  0.072      1.65    42.9
Vitamin D             551.0    NaN           NaN  ...    0.0     0.022    29.3
Vitamin E             551.0    NaN           NaN  ...  0.041      0.65    14.1
Vitamin K             551.0    NaN           NaN  ...  0.009    0.0625   166.4
Calcium               551.0    NaN           NaN  ...   38.3     101.4  1283.5
Copper                551.0    NaN           NaN  ...    0.1       0.3   668.6
Iron                  551.0    NaN           NaN  ...    1.0       2.2    21.1
Magnesium             551.0    NaN           NaN  ...   21.1      47.1   376.2
Manganese             551.0    NaN           NaN  ...    0.1       0.5   111.6
Phosphorus            551.0    NaN           NaN  ...  134.1    317.45  1385.1
Potassium             551.0    NaN           NaN  ...  223.2     456.1  3198.4
Selenium              551.0    NaN           NaN  ...   0.05     0.084  1679.0
Zinc                  551.0    NaN           NaN  ...    0.8       1.7   147.3
Nutrition Density     551.0    NaN           NaN  ...   84.1  170.6915  1337.0

[35 rows x 11 columns]

##### Data 2

In [25]:
data_2_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 319 entries, 0 to 318
Data columns (total 35 columns):
 #   Column                Non-Null Count  Dtype  
---  ------                --------------  -----  
 0   food                  319 non-null    object 
 1   Caloric Value         319 non-null    int64  
 2   Fat                   319 non-null    float64
 3   Saturated Fats        319 non-null    float64
 4   Monounsaturated Fats  319 non-null    float64
 5   Polyunsaturated Fats  319 non-null    float64
 6   Carbohydrates         319 non-null    float64
 7   Sugars                319 non-null    float64
 8   Protein               319 non-null    float64
 9   Dietary Fiber         319 non-null    float64
 10  Cholesterol           319 non-null    float64
 11  Sodium                319 non-null    float64
 12  Water                 319 non-null    float64
 13  Vitamin A             319 non-null    float64
 14  Vitamin B1            319 non-null    float64
 15  Vitamin B11           3

In [24]:
data_2_df.head()

food  Caloric Value  ...   Zinc  Nutrition Density
0                eggnog            224  ...  1.200            377.200
1            beer light             96  ...  0.044             19.456
2        beer budweiser             12  ...  0.000              2.200
3   weizenbier erdinger            220  ...  0.000             18.000
4  beer light budweiser              9  ...  0.000              1.320

[5 rows x 35 columns]

In [36]:
data_2_df.describe(include="all").T

count unique     top freq  ...     25%    50%     75%     max
food                    319    319  eggnog    1  ...     NaN    NaN     NaN     NaN
Caloric Value         319.0    NaN     NaN  NaN  ...    82.0  155.0   352.5  3190.0
Fat                   319.0    NaN     NaN  NaN  ...  0.0835    1.8     6.8   161.3
Saturated Fats        319.0    NaN     NaN  NaN  ...   0.005    0.3    2.25   396.1
Monounsaturated Fats  319.0    NaN     NaN  NaN  ...     0.0    0.4    2.45    99.9
Polyunsaturated Fats  319.0    NaN     NaN  NaN  ...     0.0    0.3    1.65   188.0
Carbohydrates         319.0    NaN     NaN  NaN  ...     4.7   18.6   49.75   390.2
Sugars                319.0    NaN     NaN  NaN  ...     0.0    0.3     6.6   291.5
Protein               319.0    NaN     NaN  NaN  ...     0.1    2.0     7.0    52.2
Dietary Fiber         319.0    NaN     NaN  NaN  ...     0.0    0.5     2.6    76.5
Cholesterol           319.0    NaN     NaN  NaN  ...     0.0    0.0     0.0   604.9
Sodium                319.0    NaN     NaN  NaN  ...   0.015  0.051  0.0985     3.0
Water                 319.0    NaN     NaN  NaN  ...    0.85   10.9    41.4   568.1
Vitamin A             319.0    NaN     NaN  NaN  ...     0.0    0.0   0.029   362.7
Vitamin B1            319.0    NaN     NaN  NaN  ...  0.0045  0.062     0.2     3.4
Vitamin B11           319.0    NaN     NaN  NaN  ...     0.0   0.03   0.073     1.1
Vitamin B12           319.0    NaN     NaN  NaN  ...     0.0    0.0   0.006   0.099
Vitamin B2            319.0    NaN     NaN  NaN  ...   0.007  0.062     0.2     2.5
Vitamin B3            319.0    NaN     NaN  NaN  ...   0.021    0.3     2.0    40.1
Vitamin B5            319.0    NaN     NaN  NaN  ...     0.0    0.1     0.5    11.8
Vitamin B6            319.0    NaN     NaN  NaN  ...   0.001  0.052     0.1     4.8
Vitamin C             319.0    NaN     NaN  NaN  ...     0.0    0.0   0.087    60.0
Vitamin D             319.0    NaN     NaN  NaN  ...     0.0    0.0     0.0    28.0
Vitamin E             319.0    NaN     NaN  NaN  ...     0.0    0.0     0.3    30.7
Vitamin K             319.0    NaN     NaN  NaN  ...     0.0    0.0   0.045     2.5
Calcium               319.0    NaN     NaN  NaN  ...    0.45   11.8   36.65   889.5
Copper                319.0    NaN     NaN  NaN  ...  0.0175  0.076     0.2   104.0
Iron                  319.0    NaN     NaN  NaN  ...   0.048    0.5     1.8    40.0
Magnesium             319.0    NaN     NaN  NaN  ...    0.55   10.1   34.35   921.6
Manganese             319.0    NaN     NaN  NaN  ...  0.0115    0.1     0.6   249.2
Phosphorus            319.0    NaN     NaN  NaN  ...     0.7   40.5   136.3  1978.9
Potassium             319.0    NaN     NaN  NaN  ...     8.9   63.5   164.9  2011.5
Selenium              319.0    NaN     NaN  NaN  ...   0.004  0.034  0.0675  1601.6
Zinc                  319.0    NaN     NaN  NaN  ...    0.05    0.3    1.15    13.3
Nutrition Density     319.0    NaN     NaN  NaN  ...  16.384   57.8   146.6  1533.5

[35 rows x 11 columns]

##### Data 3

In [26]:
data_3_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 571 entries, 0 to 570
Data columns (total 35 columns):
 #   Column                Non-Null Count  Dtype  
---  ------                --------------  -----  
 0   food                  571 non-null    object 
 1   Caloric Value         571 non-null    int64  
 2   Fat                   571 non-null    float64
 3   Saturated Fats        571 non-null    float64
 4   Monounsaturated Fats  571 non-null    float64
 5   Polyunsaturated Fats  571 non-null    float64
 6   Carbohydrates         571 non-null    float64
 7   Sugars                571 non-null    float64
 8   Protein               571 non-null    float64
 9   Dietary Fiber         571 non-null    float64
 10  Cholesterol           571 non-null    float64
 11  Sodium                571 non-null    float64
 12  Water                 571 non-null    float64
 13  Vitamin A             571 non-null    float64
 14  Vitamin B1            571 non-null    float64
 15  Vitamin B11           5

In [27]:
data_3_df.head()

food  Caloric Value    Fat  ...  Selenium   Zinc  Nutrition Density
0         nectarine             66  0.500  ...   301.500  0.000             20.735
1    kiwifruit gold             51  0.200  ...     0.003  0.077            159.686
2  prickly pear raw              8  0.072  ...     0.023  0.073             39.263
3         pineapple             45  0.100  ...    98.100  0.061             13.970
4             rowan            253  4.600  ...     0.000  1.000            176.400

[5 rows x 35 columns]

In [35]:
data_3_df.describe(include="all").T

count unique        top  ...    50%     75%       max
food                    571    571  nectarine  ...    NaN     NaN       NaN
Caloric Value         571.0    NaN        NaN  ...  145.0   406.5    6077.0
Fat                   571.0    NaN        NaN  ...    2.4   13.65     550.7
Saturated Fats        571.0    NaN        NaN  ...    0.7    4.45     672.0
Monounsaturated Fats  571.0    NaN        NaN  ...    0.6     5.1     291.1
Polyunsaturated Fats  571.0    NaN        NaN  ...    0.4     1.7      67.9
Carbohydrates         571.0    NaN        NaN  ...    0.8   13.95     134.8
Sugars                571.0    NaN        NaN  ...    0.0     0.6      97.7
Protein               571.0    NaN        NaN  ...    6.2    40.9     560.3
Dietary Fiber         571.0    NaN        NaN  ...    0.0    2.05      58.6
Cholesterol           571.0    NaN        NaN  ...    3.8  138.35   10509.0
Sodium                571.0    NaN        NaN  ...   0.08     0.2       7.0
Water                 571.0    NaN        NaN  ...   76.7  150.05    1875.9
Vitamin A             571.0    NaN        NaN  ...    0.0     0.0     134.1
Vitamin B1            571.0    NaN        NaN  ...  0.073     0.2      16.9
Vitamin B11           571.0    NaN        NaN  ...  0.058  0.0895       4.6
Vitamin B12           571.0    NaN        NaN  ...  0.024   0.063       0.5
Vitamin B2            571.0    NaN        NaN  ...   0.08     0.4      13.5
Vitamin B3            571.0    NaN        NaN  ...    1.0     5.4     124.0
Vitamin B5            571.0    NaN        NaN  ...    0.5     1.5      74.3
Vitamin B6            571.0    NaN        NaN  ...    0.2     0.6      15.8
Vitamin C             571.0    NaN        NaN  ...  0.094    2.65     541.0
Vitamin D             571.0    NaN        NaN  ...    0.0   0.047     217.6
Vitamin E             571.0    NaN        NaN  ...   0.01     0.2       8.9
Vitamin K             571.0    NaN        NaN  ...    0.0   0.066      18.3
Calcium               571.0    NaN        NaN  ...   10.5   38.45     800.8
Copper                571.0    NaN        NaN  ...    0.2    1.75    1890.0
Iron                  571.0    NaN        NaN  ...    0.8     3.6     121.2
Magnesium             571.0    NaN        NaN  ...   12.5    48.2     746.8
Manganese             571.0    NaN        NaN  ...  0.098     1.8     451.0
Phosphorus            571.0    NaN        NaN  ...   46.4   360.7    5490.0
Potassium             571.0    NaN        NaN  ...  227.5   597.0   11336.9
Selenium              571.0    NaN        NaN  ...  0.066     0.3    3308.0
Zinc                  571.0    NaN        NaN  ...    0.5     4.4      69.8
Nutrition Density     571.0    NaN        NaN  ...   57.4  153.15  1264.074

[35 rows x 11 columns]

##### Data 4

In [28]:
data_4_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 232 entries, 0 to 231
Data columns (total 35 columns):
 #   Column                Non-Null Count  Dtype  
---  ------                --------------  -----  
 0   food                  232 non-null    object 
 1   Caloric Value         232 non-null    int64  
 2   Fat                   232 non-null    float64
 3   Saturated Fats        232 non-null    float64
 4   Monounsaturated Fats  232 non-null    float64
 5   Polyunsaturated Fats  232 non-null    float64
 6   Carbohydrates         232 non-null    float64
 7   Sugars                232 non-null    float64
 8   Protein               232 non-null    float64
 9   Dietary Fiber         232 non-null    float64
 10  Cholesterol           232 non-null    float64
 11  Sodium                232 non-null    float64
 12  Water                 232 non-null    float64
 13  Vitamin A             232 non-null    float64
 14  Vitamin B1            232 non-null    float64
 15  Vitamin B11           2

In [29]:
data_4_df.head()

food  Caloric Value  ...   Zinc  Nutrition Density
0  chocolate pudding fat free            105  ...  0.300             72.400
1             tapioca pudding            143  ...  0.200            108.800
2    tapioca pudding fat free            105  ...  0.200             84.500
3                rice pudding            122  ...  0.083             27.329
4                corn pudding            328  ...  0.069             69.795

[5 rows x 35 columns]

In [34]:
data_4_df.describe(include="all").T

count unique  ...       75%     max
food                    232    232  ...       NaN     NaN
Caloric Value         232.0    NaN  ...     163.0  1010.0
Fat                   232.0    NaN  ...     4.075    95.5
Saturated Fats        232.0    NaN  ...       1.3    80.0
Monounsaturated Fats  232.0    NaN  ...     2.075    44.7
Polyunsaturated Fats  232.0    NaN  ...       1.4    51.3
Carbohydrates         232.0    NaN  ...    28.475   127.0
Sugars                232.0    NaN  ...      17.4   255.0
Protein               232.0    NaN  ...       4.0   175.0
Dietary Fiber         232.0    NaN  ...       1.6    34.0
Cholesterol           232.0    NaN  ...       0.0   184.5
Sodium                232.0    NaN  ...   0.08625     1.4
Water                 232.0    NaN  ...     209.5   499.5
Vitamin A             232.0    NaN  ...       0.0    45.0
Vitamin B1            232.0    NaN  ...     0.092    25.0
Vitamin B11           232.0    NaN  ...     0.078   177.6
Vitamin B12           232.0    NaN  ...       0.0    25.0
Vitamin B2            232.0    NaN  ...       0.1    35.0
Vitamin B3            232.0    NaN  ...     0.725    28.2
Vitamin B5            232.0    NaN  ...       0.4    25.4
Vitamin B6            232.0    NaN  ...       0.2     5.0
Vitamin C             232.0    NaN  ...     1.925  3872.0
Vitamin D             232.0    NaN  ...       0.0   170.3
Vitamin E             232.0    NaN  ...   0.09925    16.2
Vitamin K             232.0    NaN  ...   0.03325     2.4
Calcium               232.0    NaN  ...    39.475  1208.4
Copper                232.0    NaN  ...       0.4   283.2
Iron                  232.0    NaN  ...       0.9    17.3
Magnesium             232.0    NaN  ...    28.525   763.7
Manganese             232.0    NaN  ...     0.825    91.6
Phosphorus            232.0    NaN  ...      84.0  1590.6
Potassium             232.0    NaN  ...   236.175  3535.0
Selenium              232.0    NaN  ...     0.071   527.3
Zinc                  232.0    NaN  ...     0.525    11.1
Nutrition Density     232.0    NaN  ...  109.1815  3911.4

[35 rows x 11 columns]

##### Data 5

In [30]:
data_5_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 722 entries, 0 to 721
Data columns (total 35 columns):
 #   Column                Non-Null Count  Dtype  
---  ------                --------------  -----  
 0   food                  722 non-null    object 
 1   Caloric Value         722 non-null    int64  
 2   Fat                   722 non-null    float64
 3   Saturated Fats        722 non-null    float64
 4   Monounsaturated Fats  722 non-null    float64
 5   Polyunsaturated Fats  722 non-null    float64
 6   Carbohydrates         722 non-null    float64
 7   Sugars                722 non-null    float64
 8   Protein               722 non-null    float64
 9   Dietary Fiber         722 non-null    float64
 10  Cholesterol           722 non-null    float64
 11  Sodium                722 non-null    float64
 12  Water                 722 non-null    float64
 13  Vitamin A             722 non-null    float64
 14  Vitamin B1            722 non-null    float64
 15  Vitamin B11           7

In [31]:
data_5_df.head()

food  Caloric Value  ...   Zinc  Nutrition Density
0  margarine with yoghurt             88  ...  0.008             12.971
1   sunflower seed butter             99  ...  0.800             27.500
2            hazelnut oil            120  ...  0.000             13.600
3       menhaden fish oil           1966  ...  0.000            218.000
4      cod liver fish oil            123  ...  0.000             17.700

[5 rows x 35 columns]

In [33]:
data_5_df.describe(include="all").T

count unique  ...      75%     max
food                    722    722  ...      NaN     NaN
Caloric Value         722.0    NaN  ...    123.0  1966.0
Fat                   722.0    NaN  ...    4.375   218.0
Saturated Fats        722.0    NaN  ...    0.875   196.2
Monounsaturated Fats  722.0    NaN  ...      1.4    95.9
Polyunsaturated Fats  722.0    NaN  ...      1.0   115.8
Carbohydrates         722.0    NaN  ...   19.775   190.1
Sugars                722.0    NaN  ...      3.0   105.0
Protein               722.0    NaN  ...      3.7    67.9
Dietary Fiber         722.0    NaN  ...      2.8    50.2
Cholesterol           722.0    NaN  ...      0.0  1135.8
Sodium                722.0    NaN  ...      0.2    49.4
Water                 722.0    NaN  ...  109.625   690.4
Vitamin A             722.0    NaN  ...      0.0     4.1
Vitamin B1            722.0    NaN  ...  0.09575     6.3
Vitamin B11           722.0    NaN  ...    0.082     0.7
Vitamin B12           722.0    NaN  ...  0.03575     0.3
Vitamin B2            722.0    NaN  ...    0.096     1.6
Vitamin B3            722.0    NaN  ...      1.0    16.6
Vitamin B5            722.0    NaN  ...      0.5     8.0
Vitamin B6            722.0    NaN  ...      0.2     5.2
Vitamin C             722.0    NaN  ...    2.675   348.4
Vitamin D             722.0    NaN  ...      0.0   181.7
Vitamin E             722.0    NaN  ...    0.275    41.6
Vitamin K             722.0    NaN  ...    0.081     6.1
Calcium               722.0    NaN  ...   35.225   868.0
Copper                722.0    NaN  ...      0.5   348.0
Iron                  722.0    NaN  ...      1.1    29.2
Magnesium             722.0    NaN  ...     23.2   520.8
Manganese             722.0    NaN  ...      0.9   156.6
Phosphorus            722.0    NaN  ...    58.15  1309.4
Potassium             722.0    NaN  ...   191.35  4053.0
Selenium              722.0    NaN  ...    0.096  1309.5
Zinc                  722.0    NaN  ...      0.4     9.1
Nutrition Density     722.0    NaN  ...    83.45   918.8

[35 rows x 11 columns]

## Data Preparation

## Modeling

In [ ]:
import tensorflow as tf
from tensorflow.keras.models import Sequential
from tensorflow.keras.layers import Input, Dense, Dropout, BatchNormalization
from tensorflow.keras.optimizers import Adam, RMSprop, SGD
from tensorflow.keras.losses import mse, huber
from tensorflow.keras.callbacks import EarlyStopping, Callback

from sklearn.preprocessing import MinMaxScaler, StandardScaler
from sklearn.model_selection import train_test_split
from sklearn.metrics import classification_report, confusion_matrix

## Training

## Evaluation

## Prediction

## Conclusion

## Further Development